gpt_index/callbacks/schema.py

```diff
@@ -0,0 +1,47 @@
+"""Base schema for callback managers."""
+import uuid
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, Optional
+
+TIMESTAMP_FORMAT = "%m/%d/%Y, %H:%M:%S"
+
+
+class CBEventType(str, Enum):
+    """Callback manager event types."""
+
+    CHUNKING = "chunking"
+    EMBEDDING = "embedding"
+    LLM = "llm"
+    # TODO: Can we use these anywhere?
+    # QUERY = "query"
+    # RETRIEVE = "retrieve"
+    # SYNTHESIZE = "synthesize"
+    TREE = "tree"
+
+
+@dataclass
+class CBEvent:
+    """Generic class to store event information."""
+
+    event_type: CBEventType
+    payload: Optional[Dict[str, Any]] = None
+    time: str = ""
+    id: str = ""

```

Comment:
very nit: `id` is a built in python function. Use `id_` to avoid shadowing it.

Reply:
Yea good catch